src/query-parser.ts

```diff
@@ -1,15 +1,39 @@
 import { Dictionary } from "@navarik/types"
 import { SearchQuery } from "./types"
 
+const dataTypes = {
+  double: (value) => Number(value),
+  int: (value) => Number(value),
+  float: (value) => Number(value),
+  long: (value) => Number(value),
+  boolean: (value) => Boolean(value)
+}
+
+const castType = (value, type) => {
+  if (!type) {
+    return value
+  }
+  const castFunction = dataTypes[type.split(' ')[0]]
```

Comment:
oh... damn. this isn't good =) The fact that you need a split here is not a very good sign

Reply:
it should be split(',').... it's to split something like 'string, required'